src/detect-xbits.c

```diff
@@ -538,6 +542,166 @@ static int XBitsTestSig02(void)
     PASS;
 }
 
+/* Test to demonstrate redmine bug 4820 */
+static int XBitsTestSig03(void)
+{
+    DetectEngineCtx *de_ctx = NULL;
+    XBitsTestSetup();
+    de_ctx = DetectEngineCtxInit();
+    FAIL_IF_NULL(de_ctx);
+    de_ctx->flags |= DE_QUIET;
+
+    Signature *s = DetectEngineAppendSig(
+            de_ctx, "alert http any any -> any any (msg:\"TEST - No Error\")\";\
+            flow:established,to_server; http.method; content:\"GET\"; \
+            xbits:set,ET.2020_8260.1,track ip_src,expire 10,noalert; sid:1;)");
+    FAIL_IF_NULL(s);
+
+    DetectEngineCtxFree(de_ctx);
+    XBitsTestShutdown();
+    PASS;
+}
+
+/* Test to demonstrate redmine bug 4820 */
+static int XBitsTestSig04(void)
+{
+    DetectEngineCtx *de_ctx = NULL;
+    XBitsTestSetup();
+    de_ctx = DetectEngineCtxInit();
+    FAIL_IF_NULL(de_ctx);
+    de_ctx->flags |= DE_QUIET;
+
+    Signature *s =
+            DetectEngineAppendSig(de_ctx, "alert http any any -> any any (msg:\"TEST - Error\")\"; \
+            flow:established,to_server; http.method; content:\"GET\"; \
+            xbits:set,ET.2020_8260.1,noalert,track ip_src,expire 10; sid:2;)");
+    FAIL_IF_NOT_NULL(s);
+
+    DetectEngineCtxFree(de_ctx);
+    XBitsTestShutdown();
+    PASS;
+}
+
+static int XBitsTestSig05(void)
+{
+    DetectEngineCtx *de_ctx = NULL;
+    XBitsTestSetup();
+    de_ctx = DetectEngineCtxInit();
+    FAIL_IF_NULL(de_ctx);
+    de_ctx->flags |= DE_QUIET;
+
+    Signature *s = DetectEngineAppendSig(de_ctx,
+            "alert http any any -> any any (msg:\"ET EXPLOIT Possible Pulse Secure VPN RCE "
+            "Chain Stage 1 Inbound - Request Config Backup (CVE-2020-8260)\"; "
+            "flow:established,to_server; http.method; content:\"GET\"; http.uri; "
+            "content:\"/dana-admin/cached/config/config.cgi?type=system\"; fast_pattern; "
+            "xbits:set,ET.2020_8260.1,track ip_src,expire 10,noalert; "
+            "reference:url,packetstormsecurity.com/files/160619/"
+            "Pulse-Secure-VPN-Remote-Code-Execution.html; reference:cve,2020-8260; "
```

Comment:
is there a point to adding the full rule incl references and such? Seems to overdo things a bit to me.

Reply:
OK. I copied and pasted the entire rule that was failing.